contributors/guide/pull-requests.md

```diff
@@ -280,26 +277,41 @@ Each commit should have a good title line (<70 characters) and include an additi
 
  Don't squash when there are independent changes layered to achieve a single goal. For instance, writing a code munger could be one commit, applying it could be another, and adding a precommit check could be a third. One could argue they should be separate pull requests, but there's really no way to test/review the munger without seeing it applied, and there needs to be a precommit check to ensure the munged output doesn't immediately get out of date.
 
-A commit, as much as possible, should be a single logical change.
-
-## 7. KISS, YAGNI, MVP, etc.
+## 7. Commit Message Guidelines
+ 
+- Each commit should have a good title line (< 50 characters) and can include an additional description paragraph in the PR Description describing the change in more detail.
+- Wrap the body of the commit message at 72 characters and use the body to explain **What and Why** is the change and **How** it has been done.
+- It is better if the Pull Request follow the `Semantic PR Commit Guidelines` which is `{Type}: {Subject}`. **Types** can be classified as :
+    + `feat` : A PR concerning with a new feature.
+    + `fix` : A PR concerning with a bug fix.
+    + `improvement` : A PR concerning with an improvement to a current feature.
+    + `docs` : A PR which only make changes to the documentation.
+    + `refactor` : A PR concerning with a code change that neither fixes a bug nor adds a feature.
+    + `test` : A PR which adds missing tests or corrects the existing tests.
+    + `chore` : Other changes which don't modify src or test files and can be regarded as maintenance check-up for the same.
+- Additional Readings :
+    + [Git Commit](https://chris.beams.io/posts/git-commit/)
```

Comment:
Yes!  This does a good job of describing the  "what and why" distinction from "how".